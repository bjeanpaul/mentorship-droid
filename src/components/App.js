import React from 'react-native';
import { Provider } from 'react-redux';

import configureStore from '../stores/configureStore';
const store = configureStore();


console.log('----------------------------')
import { login } from '../mentor/MentorActions';
store.dispatch(login('admin', '123'));
console.log('----------------------------')


import RootNavigator from './RootNavigator';



const App = function App() {
  return (
    <Provider store={store}>
        <RootNavigator />
    </Provider>
  );
};

export default App;
