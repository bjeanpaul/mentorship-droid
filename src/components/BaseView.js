import React from 'react';
import { View, StyleSheet, StatusBar } from 'react-native';
import colors from 'src/constants/colors';


const styles = StyleSheet.create({
  view: {
    backgroundColor: colors.DEFAULT_BG,
    flex: 1,
  },
});

const BaseView = ({
  children,
  style,
  statusBarBackgroundColor = colors.DEFAULT_STATUS_BG,
}) => (
  <View style={[styles.view, style]}>
    { statusBarBackgroundColor
      ? <StatusBar backgroundColor={statusBarBackgroundColor} />
      : null
    }
    {children}
  </View>
);
BaseView.propTypes = {
  statusBarBackgroundColor: React.PropTypes.string,
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};
export default BaseView;
