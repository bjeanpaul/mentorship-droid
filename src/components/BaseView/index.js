import React from 'react';
import { View } from 'react-native';

import styles from './styles';


const BaseView = ({
  children,
  style,
}) => (
  <View style={[styles.view, style]}>
    {children}
  </View>
);


BaseView.propTypes = {
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};


export default BaseView;
