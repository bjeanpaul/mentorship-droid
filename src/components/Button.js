import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
