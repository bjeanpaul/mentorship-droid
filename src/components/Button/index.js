import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
