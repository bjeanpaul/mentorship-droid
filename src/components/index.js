import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
