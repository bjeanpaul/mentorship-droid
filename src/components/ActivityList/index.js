import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
