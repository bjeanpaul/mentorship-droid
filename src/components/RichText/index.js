import React, { Component, PropTypes } from 'react';
import { View } from 'react-native';
import mdStyles from 'react-native-simple-markdown/styles';

import { RichText as RichTextObject } from 'src/api';
import baseStyles from './styles';


class RichText extends Component {
  shouldComponentUpdate() {
  }

  getStyles() {
    return {
      ...mdStyles,
      ...baseStyles,
    };
  }

  render() {
    const styles = this.getStyles();

    return (
      <View style={styles.container} />
    );
  }
}


RichText.propTypes = {
  children: PropTypes.instanceOf(RichTextObject).isRequired,
};


export default RichText;
