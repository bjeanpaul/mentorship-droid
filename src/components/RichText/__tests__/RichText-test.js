import React from 'react';
import { Text, RichText } from 'src/components';
import { richText } from 'src/api';


describe('RichText', () => {
  it('should render typography', () => {
    // TODO get code blocks working
    expect(render(
      <RichText>{richText()(`
# h1

## h2

### h3

#### h4

##### h5

###### h6

text

[link](http://www.google.com)

*italics*
**bold**
***bold italics***
~~strikethrough~~
__underline__
__*underline italics*__
__**underline bold**__
__***underline bold italics***__
\`inlineCode\`

> block quote

- unordered
- list

1. numbered
2. list
      `)}</RichText>
    )).toMatchSnapshot();
  });

  it('should support style overrides', () => {
    const styles = {
      text: { fontSize: 100 },
    };

    expect(render(
      <RichText styles={styles}>{richText()('style overrides')}</RichText>
    )).toMatchSnapshot();
  });

  it('should support rule overrides', () => {
    const rules = styles => ({
      text: {
        react: (node, _, state) => (
          <Text key={state.key} style={[styles.text, { color: '#cc3333' }]}>
            {node.content}
          </Text>
        ),
      },
    });

    expect(render(
      <RichText rules={rules}>{richText()('rule overrides')}</RichText>
    )).toMatchSnapshot();
  });

  describe('shouldComponentUpdate', () => {
    it('should compare props via a SameValueZero comparison', () => {
      const content1 = richText()('foo');
      const content2 = richText()('foo');

      const props = {
        styles: {},
        rules: () => ({}),
        children: content1,
      };

      const el = shallow(<RichText {...props} />);

      expect(el.instance().shouldComponentUpdate({
        ...props,
        children: content1,
      })).toEqual(false);

      expect(el.instance().shouldComponentUpdate({
        ...props,
        children: content2,
      })).toEqual(true);
    });
  });
});
