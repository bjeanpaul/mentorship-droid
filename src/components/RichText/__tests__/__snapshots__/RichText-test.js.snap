exports[`RichText should render typography 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}>
    
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontFamily": "Brandon Text_medium",
          "fontSize": 24,
          "fontWeight": "400",
          "marginBottom": 25,
          "marginTop": 11,
          "textAlign": "center",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h1
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontSize": 18,
          "marginBottom": 13,
          "marginTop": 27,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h2
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontSize": 16,
          "marginBottom": 10,
          "marginTop": 20,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h3
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontSize": 15,
          "marginBottom": 7,
          "marginTop": 14,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h4
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontSize": 14,
          "marginBottom": 4,
          "marginTop": 8,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h5
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Brandon Text_bold",
          "fontWeight": "bold",
        },
        Object {
          "fontSize": 13,
          "marginBottom": 3,
          "marginTop": 6,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      h6
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "left",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      text
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "left",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
          ]
        }>
        link
      </Text>
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "left",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontStyle": "italic",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
          ]
        }>
        italics
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
          ]
        }>
        bold
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontStyle": "italic",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          bold 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          italics
        </Text>
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "line-through",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
          ]
        }>
        strikethrough
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
          ]
        }>
        underline
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "fontStyle": "italic",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          underline 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          italics
        </Text>
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          underline 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          bold
        </Text>
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "fontStyle": "italic",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
              ]
            }>
            underline 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
              ]
            }>
            bold 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
              ]
            }>
            italics
          </Text>
        </Text>
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
        ]
      }>
      
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#dfe5e6",
          "borderColor": "#dddddd",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#80979a",
          "fontFamily": "Courier",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 18,
        }
      }>
      inlineCode
    </Text>
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 20,
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#003035",
            "height": null,
            "marginRight": 15,
            "width": 3,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
              "marginBottom": 20,
              "textAlign": "left",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          block 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          quote
        </Text>
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#003035",
            "fontSize": 18,
          }
        }>
        • 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 18,
            "justifyContent": "flex-start",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          unordered
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#003035",
            "fontSize": 18,
          }
        }>
        • 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 18,
            "justifyContent": "flex-start",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 18,
                "marginBottom": 20,
                "textAlign": "left",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
              ]
            }>
            list
          </Text>
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#003035",
            "fontSize": 18,
          }
        }>
        • 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 18,
            "justifyContent": "flex-start",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          numbered
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#003035",
            "fontSize": 18,
          }
        }>
        • 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 18,
            "justifyContent": "flex-start",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
            ]
          }>
          list
          
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`RichText should support rule overrides 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "left",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "color": "#cc3333",
            },
          ],
        ]
      }>
      rule overrides
    </Text>
  </Text>
</View>
`;

exports[`RichText should support style overrides 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "left",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 100,
          },
          undefined,
        ]
      }>
      style 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 100,
          },
          undefined,
        ]
      }>
      overrides
    </Text>
  </Text>
</View>
`;
